### PR TITLE
Improve recovery for field access expression at the statement level

### LINKED ISSUE #34419: Invalid completions are provided for chained invocation
**Description:**
Consider the following source snippet where the shown completions are invalid.

```
match command.toLowerAscii() {
        GREETING => {
            io:println("GREETING");
            io:println(getServerGreeting(mFields.get(ProfileName)));
        }
        REGISTER => {
            io:println("REGISTER");
            io:println(storeData(mFields));
        }
        UNREGISTER => {
            io:println("UNREGISTER");
            io:println(removeUser(mFields.get(WaId)));
        }
        SUBSCRIBE => {
            (regex:replaceFirst(messageBody, command, "")).<cursor>
            io:println("SUBSCRIBE");
            io:println(subscribeToTeam(mFields.get(WaId), command));
        }
        UNSUBSCRIBE => {
            io:println("UNSUBSCRIBE");
            io:println(removeUser(mFields.get(WaId)));
        }
    } 
```

Screenshot 2021-12-21 at 08 48 26

**Steps to reproduce:**
Get the completions at the given cursor postion

**Affected Versions:**
SwanLake-Beta6 at least

### PR DESCRIPTION
**Pull request by rdulmina**
## Purpose
Improve recovery for field access expression at the statement level

Fixes #34419

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-parser/src/test/resources/statements/assignment-stmt/assignment_stmt_assert_13.json [hunk 1]
```diff
@@ -1,122 +1,90 @@
...
-        "ERROR_MISSING_EQUAL_TOKEN"
+        "ERROR_MISSING_SEMICOLON_TOKEN"
       ]
-    },
-    {
-      "kind": "SIMPLE_NAME_REFERENCE",
-      "children": [
-        {
-          "kind": "IDENTIFIER_TOKEN",
-          "value": "d"
```
Comment (lochana-chathura) [inline compiler/ballerina-parser/src/test/resources/statements/assignment-stmt/assignment_stmt_assert_13.json:114]: Looks like `a + b + c  d;` recovers `as a + b + c MISSING[;]` and remaining part not included in the json. Shall we use `testFile()` for this test case?

**Review by lochana-chathura — COMMENTED** [review]
LGTM. Added a minor suggestion.

File: compiler/ballerina-parser/src/test/resources/statements/assignment-stmt/assignment_stmt_assert_13.json (hunk 1 above)
Comment (rdulmina) [reply compiler/ballerina-parser/src/test/resources/statements/assignment-stmt/assignment_stmt_assert_13.json:114]: Updated

**Review by lochana-chathura — APPROVED** [review]

**Review by gimantha — APPROVED** [review]
LGTM. Tested and verified

**Pull request merged**